### PR TITLE
Added weights to how the intrigue game mode gets chosen

### PR DESCRIPTION
**Pull request by Gonzako**
[DISCUSSION][BALANCE][INPUT WANTED]
## About the PR 

This PR tweaks how intrigue handles game mode selection. 

## Why's this needed? 

I noticed that playing on RP the chance to become a traitor is fairly rare due to both the reduced antag count and weighting towards monster antags. While monster antags are well thought out, being able to get traitor or spy thief opens you more options than having a consistent set of powers. This PR divides the game modes into 3 categories:

- All Human antagonists:
      These are your traitors, spy thiefs and conspirators. Due to conspiracy most of the time devolving into hard mode traitor with a dissapointment I chose to bring it down in favour of traitor. 
- All non-human antagonists:
      These are pretty much un-changed.
- Mixed:
      This gamemode has been given it's own category due to (in my opinion) playing best to SS13s strenghs of not being able to know what the game can throw at you. This game mode has also been tweaked in favour of traitors to allow for an even split of human antagonists and alien antagonists.  

I chose to try and have a 30% split between these 3 categories. 

## Changelog

```changelog
(u)Gonzako
(*)How intrigue chooses game mode has been tweaked, traitors should be slightly more common while aliens should invade the station a bit less frequently. 
```

### CODE REVIEW

**Review by pali6 — COMMENTED** [review]
`prob(100)` is unnecessary as that is the default

**Comment by Gonzako** [comment]
> `prob(100)` is unnecessary as that is the default

Sorry, I'd have deleted that. I played with the numbers a bit to see something I found interesting

File: code/datums/gamemodes/mixed_rp.dm [hunk 1]
```diff
@@ -2,8 +2,10 @@
 	name = "mixed (mild)"
 	config_tag = "mixed_rp"
 	latejoin_antag_compatible = 1
-	latejoin_antag_roles = list(ROLE_TRAITOR, ROLE_CHANGELING, ROLE_VAMPIRE, ROLE_WRESTLER, ROLE_ARCFIEND)
-	traitor_types = list(ROLE_TRAITOR, ROLE_CHANGELING, ROLE_VAMPIRE, ROLE_SPY_THIEF, ROLE_ARCFIEND)
+	 //went with a trivial solution of adding more identical items to the list
+	 //Input needed here
+	latejoin_antag_roles = list(ROLE_TRAITOR, ROLE_CHANGELING, ROLE_VAMPIRE, ROLE_WRESTLER, ROLE_ARCFIEND, ROLE_TRAITOR)
+	traitor_types = list(ROLE_TRAITOR, ROLE_CHANGELING, ROLE_VAMPIRE, ROLE_SPY_THIEF, ROLE_ARCFIEND, ROLE_TRAITOR)
```
Comment (zjdtmkhzt) [inline code/datums/gamemodes/mixed_rp.dm:8]: If you're going the route of adding a thing multiple times for higher chance, wouldn't it be more readable to have the same type next to each other, so you can see how often it is in the list more easily?

Comment (zjdtmkhzt) [reply code/datums/gamemodes/mixed_rp.dm:8]: Also, typo in the word "frequently" in changelog, I think?

Comment (Gonzako) [reply code/datums/gamemodes/mixed_rp.dm:8]: Hi! I chose to add them to the end as it's the least destructive change you'd do to a list. I really dont know how said array is used so I chose the method less prone to randomly break stuff somewhere else.

Typo also fixed.

Comment (pali6) [inline code/datums/gamemodes/mixed_rp.dm:8]: Changing to an associative list and using weighted_pick proc would be a solution

File: code/datums/gameticker.dm [hunk 2]
```diff
@@ -117,7 +117,7 @@ var/global/current_state = GAME_STATE_WORLD_INIT
 	switch(master_mode)
 		if("random","secret") src.mode = config.pick_random_mode()
 		if("action") src.mode = config.pick_mode(pick("nuclear","wizard","blob"))
-		if("intrigue") src.mode = config.pick_mode(pick("mixed_rp", "traitor","changeling","vampire","conspiracy","spy_theft","arcfiend", prob(50); "extended"))
+		if("intrigue") src.mode = config.pick_mode(pick(prob(300);"mixed_rp", prob(150); "traitor","changeling","vampire", prob(50); "conspiracy", "spy_theft","arcfiend", prob(50); "extended"))
```
Comment (pali6) [inline code/datums/gameticker.dm:120]: Overall admin opinion seems to be that traitor could be at 200.

Comment (pali6) [reply code/datums/gameticker.dm:120]: Also probably changeling at 75 or so as changelings tend to make people salty a lot.

File: code/mob/new_player.dm [hunk 3]
```diff
@@ -667,8 +667,19 @@ a.latejoin-card:hover {
 							break
 
 					var/bad_type = null
-					if (islist(ticker.mode.latejoin_antag_roles) && length(ticker.mode.latejoin_antag_roles))
-						bad_type = pick(ticker.mode.latejoin_antag_roles)
+					if (islist(ticker.mode.latejoin_antag_roles) && length(ticker.mode.latejoin_antag_roles)){
+
+						//Another one I need input on
+						if(ticker.mode.latejoin_antag_roles[ROLE_TRAITOR] != null)
```
Comment (Gonzako) [inline code/mob/new_player.dm:673]: Is there any simple way of checking if a list is assosiative or not?

File: code/datums/gameticker.dm (hunk 2 above)
Comment (Gonzako) [reply code/datums/gameticker.dm]: Input written down

**Review by frawhst — APPROVED** [review]
Weights look good to me

**Pull request merged**